src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/factory/generator/flow/JointRuleGenerator.java

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.rulemanager.factory.generator.flow;
+
+import org.openkilda.model.Switch;
+import org.openkilda.rulemanager.SpeakerCommandData;
+import org.openkilda.rulemanager.factory.RuleGenerator;
+
+import lombok.Builder;
+
+import java.util.List;
+
+@Builder
+public class JointRuleGenerator implements RuleGenerator {
+    protected final RuleGenerator firstGenerator;
+    protected final RuleGenerator secondGenerator;
+
+
+    @Override
+    public List<SpeakerCommandData> generateCommands(Switch sw) {
+        List<SpeakerCommandData> commands = firstGenerator.generateCommands(sw);
+        commands.addAll(secondGenerator.generateCommands(sw));
```

Comment:
Are this two lines of code really needs a separate class? Maybe we can inline them?